pkgs/development/node-packages/default.nix

```diff
@@ -60,6 +60,42 @@ let
       nativeBuildInputs = drv.nativeBuildInputs or [] ++ [ pkgs.psc-package self.pulp ];
     });
 
+    mirakurun = super.mirakurun.override rec {
+      buildInputs = [ pkgs.which pkgs.v4l_utils ];
+      postInstall = let
+        # The original mirakurun command uses PM2 to manage the Mirakurun server.
+        # However, we invoke the server directly and let systemd manage it to
+        # avoid complication. This is okay since no features unique to PM2 is
+        # currently being used.
+        wrapperScript = pkgs.writeScript "mirakurun" ''
+          #!${stdenv.shell}
+
+          cd @out@/lib/node_modules/mirakurun
+
+          export PATH="${pkgs.lib.makeBinPath buildInputs}:$PATH"
+
+          # the cli flags are taken from package.json
+          exec ${nodejs}/bin/node --max_old_space_size=512 lib/server.js
+        '';
+      in
+      ''
+        cp ${wrapperScript} $out/bin/mirakurun
```

Comment:
You can use `makeWrapper` instead, it supports `--run` to run a `cd` command beforehand, `--prefix` for adding to `PATH` and `--add-flags` for adding flags. Can also use `$out` directly then without having to substitute it

Reply:
Fixed